src/core/openthread-core-default-config.h

```diff
@@ -1248,6 +1248,52 @@
 #define OPENTHREAD_CONFIG_NCP_SPINEL_RESPONSE_QUEUE_SIZE 15
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_NCP_ENABLE_POWER_STATE_SAVE
+ *
+ * Define to 1 to enable NCP power state control.
+ *
+ * NCP power state would allow the NCP to be put into different power state (e.g., deep-sleep, offline). Please see
+ * `spinel.h` and `spinel_power_state_t` definition.
+ *
+ * When enabled, `OPENTHREAD_CONFIG_NCP_PLAT_GET_POWER_STATE_FUNC` and `OPENTHREAD_CONFIG_NCP_PLAT_GET_POWER_STATE_FUNC`
+ * must be also provided.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_NCP_ENABLE_POWER_STATE_SAVE
+#define OPENTHREAD_CONFIG_NCP_ENABLE_POWER_STATE_SAVE 0
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_NCP_PLAT_GET_POWER_STATE_FUNC
+ *
+ * Specifies the name of the platform function/macro used to get currently set NCP power state.
+ *
+ * Applicable only if `OPENTHREAD_CONFIG_NCP_ENABLE_POWER_STATE_SAVE` is set to 1 and enabled.
+ *
+ * The function/macro format should follow `uint8_t GetPowerState(void)` where the returned value matches the
+ * `spinel_power_state_t` enumeration from `spinel.h`
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_NCP_PLAT_GET_POWER_STATE_FUNC
+#define OPENTHREAD_CONFIG_NCP_PLAT_GET_POWER_STATE_FUNC /* empty */
```

Comment:
why have you opted for #defined functions here rather than adding new API's otPlat...?